File: 07_02_Greedy/92_3_minimize_maximum_pair_sum_in_array/szzng.js
// Runtime 348 ms Beats 97.37%
// Memory 56.2 MB Beats 98.3%

var minPairSum = function(nums) {
    nums.sort((a,b)=>a-b)
    const numLength = nums.length
    const halfLength = numLength/2
    var maxSum = 0

    for (var i=0; i<halfLength; i++)
        maxSum = Math.max(nums[i] + nums[numLength-1-i] , maxSum)
    return maxSum
};

File: 07_02_Greedy/92_3_minimize_maximum_pair_sum_in_array/szzng.py
# Runtime 1188 ms Beats 80.6%
# Memory 27.8 MB Beats 50.51%

from typing import List


class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()
        half_len = len(nums)//2
        max_sum = 0

        for i in range(half_len):
            max_sum = max(max_sum, nums[i]+nums[-1*(i+1)])
        return max_sum
